src/lib.rs

```diff
@@ -246,33 +202,9 @@ pub fn create_asset(
     })
 }
 
-pub async fn import_asset(
-    rgb_tokens_descriptor: Option<&str>,
-    contract_id: Option<&str>,
-    genesis: Option<&str>,
-    node_url: Option<String>,
-) -> Result<ThinAsset> {
-    match genesis {
-        Some(genesis) => {
-            info!("Getting asset by genesis:", genesis);
-            get_asset_by_genesis(genesis)
-        }
-        None => match (contract_id, rgb_tokens_descriptor) {
-            (Some(contract_id), Some(rgb_tokens_descriptor)) => {
-                info!("Getting asset by contract id:", contract_id);
-                let wallet = get_wallet(rgb_tokens_descriptor, None)?;
-                synchronize_wallet(&wallet).await?;
-                let unspent = wallet.list_unspent().unwrap_or_default();
-                let asset = get_asset_by_contract_id(contract_id, unspent, node_url).await;
-                info!(format!("asset: {asset:?}"));
-                match asset {
-                    Ok(asset) => Ok(asset),
-                    Err(e) => Err(format_err!("Server error: {e}")),
-                }
-            }
-            _ => Err(format_err!("Error: Unknown error in import_asset")),
-        },
-    }
+pub fn import_asset(genesis: &str) -> Result<ThinAsset> {
```

Comment:
This method can only get the asset by full genesis, but the genesis should be stored via lambda keyed by rgb id, and retrieved so it can be passed to this method by the client. They should be two separate operations, so that the contents can be encrypted end-to-end. Eventually all RGB methods will be run in WASM or clientside, the RGB lambdas are temporary, but the storage lambdas are permanent.

We should have a separate method that processes an entire consignment and returns all assets and balances. I haven't worked on that yet, but will do after I get transfers working.

Does that all make sense / sound good?

Reply:
It makes sense, of course.